internal/hostnetwork/ovs_bridge.go

```diff
@@ -0,0 +1,356 @@
+}
+
+// ensureOVSBridgeAndAttachWithClient ensures an OVS bridge exists and attaches ifaceName as a port.
+// This version accepts a client parameter for testing.
+func ensureOVSBridgeAndAttachWithClient(ctx context.Context, ovs libovsclient.Client, bridgeName, ifaceName string) error {
+	// Cache for indexed operations
+	_, err := ovs.Monitor(ctx,
+		ovs.NewMonitor(
+			libovsclient.WithTable(&OpenVSwitch{}),
+			libovsclient.WithTable(&Bridge{}),
+			libovsclient.WithTable(&Port{}),
+			libovsclient.WithTable(&Interface{}),
+		),
+	)
+	if err != nil {
+		return fmt.Errorf("failed to setup monitor: %w", err)
+	}
+
+	bridgeUUID, err := ensureBridge(ctx, ovs, bridgeName)
+	if err != nil {
+		return fmt.Errorf("failed to ensure OVS bridge %q exists: %w", bridgeName, err)
+	}
+
+	// Create the bridge management port (internal port) so the bridge appears as a Linux interface
+	if err := ensureInternalPortForBridge(ctx, ovs, bridgeUUID, bridgeName); err != nil {
+		return fmt.Errorf("failed to create internal port for bridge %q: %w", bridgeName, err)
+	}
+
+	if err := ensurePortAttachedToBridge(ctx, ovs, bridgeUUID, ifaceName); err != nil {
+		return fmt.Errorf("failed to attach veth %q to bridge %q: %w", ifaceName, bridgeName, err)
+	}
+
+	// Bring the OVS bridge interface UP so it can be used as a master for macvlan/ipvlan
+	bridge, err := netlink.LinkByName(bridgeName)
+	if err != nil {
+		return fmt.Errorf("failed to get OVS bridge %q interface: %w", bridgeName, err)
+	}
+	if err := netlink.LinkSetUp(bridge); err != nil {
+		return fmt.Errorf("failed to bring OVS bridge %q UP: %w", bridgeName, err)
+	}
+	slog.Debug("OVS bridge interface is UP", "name", bridgeName)
+
+	return nil
+}
+
+func ensureBridge(ctx context.Context, ovs libovsclient.Client, bridgeName string) (string, error) {
+	br := &Bridge{Name: bridgeName}
+	err := ovs.Get(ctx, br)
+	if err == nil {
+		return br.UUID, nil
+	}
+
+	namedUUID := "new_bridge"
+	br = &Bridge{
+		UUID:        namedUUID,
+		Name:        bridgeName,
+		ExternalIds: map[string]string{"created-by": "openperouter"},
+	}
+
+	insertOp, err := ovs.Create(br)
+	if err != nil {
+		return "", fmt.Errorf("failed to create bridge insert operation: %w", err)
+	}
+
+	ovsRow := &OpenVSwitch{}
+	mutateOp, err := ovs.WhereCache(func(*OpenVSwitch) bool { return true }).
+		Mutate(ovsRow, model.Mutation{
+			Field:   &ovsRow.Bridges,
+			Mutator: ovsdb.MutateOperationInsert,
+			Value:   []string{namedUUID},
+		})
+	if err != nil {
+		return "", fmt.Errorf("failed to create mutate operation: %w", err)
+	}
+
+	operations := append(insertOp, mutateOp...)
+	reply, err := ovs.Transact(ctx, operations...)
+	if err != nil {
+		return "", fmt.Errorf("transaction failed: %w", err)
+	}
+
+	_, err = ovsdb.CheckOperationResults(reply, operations)
+	if err != nil {
+		return "", fmt.Errorf("operation failed: %w", err)
+	}
+
+	realUUID := reply[0].UUID.GoUUID
+	slog.Debug("created OVS bridge", "name", bridgeName, "UUID", realUUID)
```

Comment:
I'd say we only care about user provided ovs bridges, or do you see a use case where it'd make sense for the perouter to create the bridge?

Reply:
The API has an auto-create attribute ...

If there's a reason to create linux bridges, I'd say there's a reason to create ovs bridges ... 

e.g. OVS CNI (it requires an existing bridge, which *something* must create).